internal/host/host.go

```diff
@@ -320,23 +339,32 @@ func (m *Manager) RefreshStatus(ctx context.Context, h *models.Host, db *gorm.DB
 	if err != nil {
 		return err
 	}
-	conditions, validationsResults, err := m.rp.preprocess(vc)
+	conditions, newValidationRes, err := m.rp.preprocess(vc)
 	if err != nil {
 		return err
 	}
-	if err = m.reportValidationStatusChanged(ctx, vc, h, validationsResults); err != nil {
+	currentValidationRes, validationsChanged, err := m.checkValidationChanged(ctx, h, newValidationRes)
+	if err != nil {
 		return err
 	}
+	if validationsChanged {
```

Comment:
I think we should add a comment on why we are doing this before updating the DB

Reply:
I think is is pretty clear by the fact that it gets both, current validations and new validations don't you think so?

Reply:
everyone should understand why we split the update DB & report metrics and we are not doing it at the same time - I'm afraid that someone will merge the code and it will cause us bugs again.

Reply:
I have changed the order and added a comment.